.github/workflows/publish-docker.yml

```
@@ -37,8 +37,10 @@ jobs:
 
       - name: Set commit short hash
         id: setHash
+        # github.event.pull_request.head.sha is the latest commit for a pull request
+        # github.sha is the latest commit for a merge to main (but is the merge commit on PRs)
```

Super nit: the difference here is the event that triggered the workflow (push vs pull request), not the type of commit.
https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

```suggestion
        # github.sha is the latest commit for a push to main (but is the merge commit on PRs)
```